addons/iap_mail/static/src/js/services/iap_notification_service.js

```diff
@@ -0,0 +1,42 @@
+/** @odoo-module **/
+
+import { Markup } from 'web.utils';
+import { registry } from "@web/core/registry";
+
+export const iapNotificationService = {
+    dependencies: ["notification"],
+
+    start(env, { notification }) {
+        env.bus.on("WEB_CLIENT_READY", null, async () => {
+            const legacyEnv = owl.Component.env;
+            legacyEnv.services.bus_service.onNotification(this, (notifications) => {
+                for (const { payload, type } of notifications) {
+                    if (type === 'iap_notification') {
+                        displayIapNotification(payload);
+                    }
+                }
+            });
+            legacyEnv.services.bus_service.startPolling();
+        });
+
+        /**
+         * Displays the IAP notification on user's screen
+         */
+        function displayIapNotification(notif) {
+            if (notif.url) {
+                const message = Markup`<a class='btn btn-link' href=${notif.url} target='_blank' ><i class='fa fa-arrow-right'></i> ${env._t("Buy more credits")}</a>`;
+                notification.add(message, {
+                    type: notif.error_type,
+                    messageIsHtml: true,
+                    title: notif.title
+                });
+            } else {
+                notification.add(notif.title, {
+                    type: notif.error_type,
+                });
+            }
```

Comment:
As the handling in if/else is completely different, I would consider 2 separate method with 2 different notification type from the server, and move the condition earlier in the chain.
It's generally not a good idea to make a condition on payload, it's easier to manage if the only condition is on type.